lib/blacklight/communication.rb

```diff
@@ -0,0 +1,35 @@
+require "fileutils"
+require "securerandom"
+require_relative "exceptions"
+
+module Blacklight
+  def self.iterate_course(xml_data, course)
```

Comment:
I personally would rather these methods were pure. Would it make sense to return a new course with these values set?